### PR TITLE
[bugfix] backport from 3.0 Fix unit of sizes when reading a SLD file (fixes #8978)

### PR DESCRIPTION
**Pull request by luipir**
## Description
This PR is the backport of #4432 that add correct support for SLD "uom" parameters. e.g. unit size specified in the SLD.

(this PR is the rebased version of the closed https://github.com/qgis/QGIS/pull/4731)

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

- [x] Commit messages are descriptive and explain the rationale for changes
- [x] Commits which fix bugs include `fixes #11111` in the commit message next to the description
- [ ] Commits which add new features are tagged with `[FEATURE]` in the commit message
- [ ] Commits which change the UI or existing user workflows are tagged with `[needs-docs]` in the commit message and containt sufficient information in the commit message to be documented
- [x] I have read the [QGIS Coding Standards](https://docs.qgis.org/testing/en/docs/developers_guide/codingstandards.html) and this PR complies with them
- [ ] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [x] New unit tests have been added for core changes
- [x] I have run [the `scripts/prepare-commit.sh` script](https://github.com/qgis/QGIS/blob/master/.github/CONTRIBUTE.md#contributing-to-qgis) before each commit

### CODE REVIEW

**Comment by luipir** [comment]
@aaime @pblottiere can you give a review of this backport?
this is exaclty the same PR in https://github.com/qgis/QGIS/pull/4731 already checked by @aaime

**Comment by luipir** [comment]
travis failed on ProcessingToolTest with something not related with this PR but with OGR tests
test_ogrLayerName

======================================================================
FAIL: test_ogrLayerName (__main__.VectorTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/qgis/QGIS/python/plugins/processing/tests/ToolsTest.py", line 81, in test_ogrLayerName
    self.assertEqual(name, 'a')
AssertionError: 'b' != 'a'

**Comment by pblottiere** [comment]
Thank you @luipir! LGTM!

File: src/core/symbology-ng/qgsellipsesymbollayerv2.cpp [hunk 1]
```diff
@@ -490,6 +490,10 @@ QgsSymbolLayerV2* QgsEllipseSymbolLayerV2::createFromSld( QDomElement &element )
   if ( !QgsSymbolLayerV2Utils::wellKnownMarkerFromSld( graphicElem, name, fillColor, borderColor, borderStyle, borderWidth, size ) )
     return nullptr;
 
+  QString uom = element.attribute( QString( "uom" ), "" );
```
Comment (pblottiere) [inline src/core/symbology-ng/qgsellipsesymbollayerv2.cpp]: Looking back, *uom* could be a `const QString` here... It's also true for the others.

Comment (luipir) [reply src/core/symbology-ng/qgsellipsesymbollayerv2.cpp]: tnx

**Comment by luipir** [comment]
I really don't understand why travins fail on ProcessingToolsTest (and gdal related stuffs) @pblottiere any idea?

**Comment by luipir** [comment]
My local tests with clean release-2_18 (e.g. without this PR) give me the same test fail but for different reason => the test is broken but not related with this PR.

2: ======================================================================
2: ERROR: test_ogrLayerName (__main__.VectorTest)
2: ----------------------------------------------------------------------
2: Traceback (most recent call last):
2:   File "/home/ginetto/PROGRAMMING/QGIS/QGIS-2.18/python/plugins/processing/tests/ToolsTest.py", line 74, in test_ogrLayerName
2:     linkTestfile('geom_data.csv', 'a.csv')
2:   File "/home/ginetto/PROGRAMMING/QGIS/QGIS-2.18/python/plugins/processing/tests/ToolsTest.py", line 71, in linkTestfile
2:     os.link(os.path.join(dataFolder, f), os.path.join(tmpdir, t))
2: OSError: [Errno 18] Invalid cross-device link

**Comment by luipir** [comment]
the same test ProcessingToolTests fail also in some PR on relase-2_14. Any idea @m-kuhn ?

**Comment by pblottiere** [comment]
I just tried on a fresh *release-2_18* build and indeed the same test is failing. However, I have a slightly different error:

````
2: ======================================================================
2: FAIL: test_ogrLayerName (__main__.VectorTest)
2: ----------------------------------------------------------------------
2: Traceback (most recent call last):
2:   File "/home/blottiere/devel/packages/oslandia/QGIS/QGIS_pbl_218/python/plugins/processing/tests/ToolsTest.py", line 81, in test_ogrLayerName
2:     self.assertEqual(name, 'a')
2: AssertionError: 'invalid-uri' != 'a'
````

**Comment by alexbruy** [comment]
This test creates hadlinks, probably it fails because temp directory is on another system. If I remember corrctly this test was added some time ago by @strk (but I can be wrong), probably he can explain where is the problem better.

**Comment by luipir** [comment]
change:
https://github.com/qgis/QGIS/blob/release-2_18/python/plugins/processing/tests/ToolsTest.py#L71
form os.link to os.symlink solve the test fail locally in my device configuration
@pblottiere can you check if it works for you?
BTW this is not probably the error origin in travis

**Comment by pblottiere** [comment]
> can you check if it works for you?

Nope, still the same error:

````
2: ======================================================================
2: FAIL: test_ogrLayerName (__main__.VectorTest)
2: ----------------------------------------------------------------------
2: Traceback (most recent call last):
2:   File "/home/blottiere/devel/packages/oslandia/QGIS/QGIS_pbl_218/python/plugins/processing/tests/ToolsTest.py", line 81, in test_ogrLayerName
2:     self.assertEqual(name, 'a')
2: AssertionError: 'invalid-uri' != 'a'
````

**Comment by luipir** [comment]
wonderful Three environments three different results :(

**Comment by luipir** [comment]
patched the test at least for my drive configuration case in : https://github.com/qgis/QGIS/pull/5151

**Comment by luipir** [comment]
@pblottiere can you merge pelase if review it's ok? fail depends on something with travis config and more PR (at least 4) with the same travis fail has been already merged

**Comment by Gustry** [comment]
@luipir Next release of 2.18 is next Friday, but as I could read in another PR, Paul is on holiday ( cc @haubourg )

**Comment by luipir** [comment]
tnx @Gustry for the clarification

**Comment by luipir** [comment]
check if fix https://github.com/qgis/QGIS/pull/5176 fix the failing travis

**Pull request merged**